### ISSUE TITLE
Global variable initializer type does not match global variable type

### ISSUE THREAD

**Issue by M1ke** [protocolbuffers/protobuf#7587]
When trying to compile a project that uses protobuf as part of react-native-firebase the following compile error occurs with GBPApi.pbobjc.m

```
CompileC /Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/Protobuf.build/Objects-normal/armv7/GPBApi.pbobjc.o Protobuf/objectivec/GPBApi.pbobjc.m normal armv7 objective-c com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/m1ke/app/ios/Pods
    export LANG=en_US.US-ASCII
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -arch armv7 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu11 -fobjc-arc -fobjc-weak -fmodules -fmodules-cache-path=/Users/m1ke/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/m1ke/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -Os -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -DPOD_CONFIGURATION_RELEASE=1 -DCOCOAPODS=1 -DPOD_CONFIGURATION_RELEASE=1 -DGPB_USE_PROTOBUF_FRAMEWORK_IMPORTS=1 -DNS_BLOCK_ASSERTIONS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.1.sdk -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -miphoneos-version-min=7.0 -g -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wno-semicolon-before-method-body -Wunguarded-availability -fembed-bitcode -iquote /Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/Protobuf.build/Protobuf-generated-files.hmap -I/Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/Protobuf.build/Protobuf-own-target-headers.hmap -I/Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/Protobuf.build/Protobuf-all-target-headers.hmap -iquote /Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/Protobuf.build/Protobuf-project-headers.hmap -I/Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/BuildProductsPath/Release-iphoneos/Protobuf/include -I/Users/m1ke/app/ios/Pods/Headers/Private -I/Users/m1ke/app/ios/Pods/Headers/Private/Protobuf -I/Users/m1ke/app/ios/Pods/Headers/Public -I/Users/m1ke/app/ios/Pods/Headers/Public/Protobuf -I/Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/Protobuf.build/DerivedSources/armv7 -I/Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/Protobuf.build/DerivedSources -F/Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/BuildProductsPath/Release-iphoneos/Protobuf -fno-objc-arc -include /Users/m1ke/app/ios/Pods/Target\ Support\ Files/Protobuf/Protobuf-prefix.pch -MMD -MT dependencies -MF /Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/Protobuf.build/Objects-normal/armv7/GPBApi.pbobjc.d --serialize-diagnostics /Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/Protobuf.build/Objects-normal/armv7/GPBApi.pbobjc.dia -c /Users/m1ke/app/ios/Pods/Protobuf/objectivec/GPBApi.pbobjc.m -o /Users/m1ke/Library/Developer/Xcode/DerivedData/myapp-adkwfvzyfxkzbfcxhbgbxeqzybrb/Build/Intermediates.noindex/ArchiveIntermediates/myapp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/Protobuf.build/Objects-normal/armv7/GPBApi.pbobjc.o

Global variable initializer type does not match global variable type!
%struct.GPBObjcClass_t* @"OBJC_CLASS_$_GPBMethod"
Global variable initializer type does not match global variable type!
%struct.GPBObjcClass_t* @"OBJC_CLASS_$_GPBMixin"
fatal error: error in backend: Broken module found, compilation aborted!
clang: error: clang frontend command failed with exit code 70 (use -v to see invocation)
Apple LLVM version 10.0.0 (clang-1000.11.45.5)
Target: arm-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

This is running under Xcode 10.3, Firebase/Core pod at version 6.14. I am reporting here as the error is in the Protobuf package rather than Firebase. Protobuf pod is 3.12.0, I believe that matches a version constraint for Firebase Messaging.

Googling various parts of this error brings up nothing but I imagine I've tried searching the wrong bits; however not being a C developer its beyond me to try and work out what's wrong here.

**Comment by M1ke** [comment]
I resolved this by downgrading from 3.12 to 3.11. Not sure if that's the appropriate fix, but it worked in my case. Still interested to see if anybody else experiences this as it is surprising to get an error in a library that it seems nobody has reported before.

**Comment by esteves-esta** [comment]
Thank you @M1ke, I am working in a Ionic 3 project that had the same error and your solution worked :smile:

**Comment by RAJKUMARRRR** [comment]
I was also facing same issue, when run on sim and device it worked fine but while Archiving faced this issue, solved it by downgrading Protobuf  from 3.12 to 3.11, thanks @M1ke

**Comment by LeungMingFai** [comment]
Same here. Downgrading to 3.11 works. Thanks!
